src/Symfony/Component/Form/Extension/Core/DataTransformer/ValueToDuplicatesTransformer.php

```diff
@@ -62,10 +64,13 @@ public function reverseTransform($array)
 
         $result = current($array);
         $emptyKeys = array();
+        $compare = is_callable($this->compare) ? $this->compare : function ($value1, $value2) {
+            return $value1 === $value2;
+        };
 
         foreach ($this->keys as $key) {
             if (isset($array[$key]) && '' !== $array[$key] && false !== $array[$key] && array() !== $array[$key]) {
-                if ($array[$key] !== $result) {
+                if (!$compare($result, $array[$key], $key)) {
```

Comment:
In the previous version (before this PR), it did not matter, as it was a simple string comparison. I just kept the order without really looking into it. But as the comparing happens on every field, it is interesting to know what field is being compared (the third parameter), and to at least have some more logical order in the values that are being compared. The $result variable is the first field, which eventually will deliver the end result when all checks are successful. The $array[$key] variable will be the first field, then the second field. It certainly is debatable to which order is preferred, but I thought that having at least the first field always be in the first spot might be more intuitive?